### PR TITLE
Admin metrics v2

### PR DESCRIPTION
**Pull request by tsun812**
Developed with @javtran 

#### Project table: 
- A sortable list of  all projects with core metrics (id,  name, owner, discoverable, archived, virtual, date created, date last modified).
- Clicking on a project will take you to the existing project admin page.
- Sort projects by name, newest/oldest date, # of challenges, default.
- Hide undiscoverable/archived projects with toggle.
- Filter virtual only projects with toggle.
- Search to filter by project name.
- Filter projects by date created.
- Export table as CSV.
- Customizable paginations(default to 50).

#### User table: 
- A sortable list of  all users with core metrics (id, name, score, date created, date last active).
- Sort users by name, newest/oldest date, score, default.
- Search to filter by osm username.
- Filter users by date created.
- Export table as CSV.
- Customizable paginations(default to 50).

#### New features/changes to challenge table: 
- Filter challenges by date created.
- Hide undiscoverable/archived challenges with toggle.
- Display project by name.

### CODE REVIEW

File: src/components/Navbar/Navbar.js [hunk 1]
```diff
@@ -360,6 +362,15 @@ const MobileNav = props => (
...
         </li>
+        {
+          props.isSuperUser
+            ? <li>
+                <NavLink to="/superadmin" onClick={props.closeMobileMenu}>
+                  <FormattedMessage {...messages.superAdmin} />
+                </NavLink>
+              </li>
+            : null
+        }
```
Comment (jschwarz2030) [inline src/components/Navbar/Navbar.js:373]: didn't we add this in v1?  why is v1 included in this PR?

Comment (tsun812) [reply src/components/Navbar/Navbar.js:373]: I don't think we have a navbar specifically for mobile in v1.

File: src/components/HOCs/WithSearchRoute/WithSearchRoute.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ export const WithSearchRoute = function(WrappedComponent, searchGroup) {
       this.props.setChallengeSearchMapBounds(bounds, fromUserAction)
 
       const locationFilter = _get(this.props, `currentSearch.${searchGroup}.filters.location`)
-      updateBoundsOnRoute(this.props, searchGroup, bounds, locationFilter, true)
+      updateBoundsOnRoute(this.props, searchGroup, bounds, locationFilter, true)
```
Comment (jschwarz2030) [inline src/components/HOCs/WithSearchRoute/WithSearchRoute.js:113]: fix these spaces please.  use prettier to highlight and format your new code.

File: src/components/ChallengePane/ChallengePane.js [hunk 3]
```diff
@@ -105,6 +106,14 @@ export class ChallengePane extends Component {
 
   resetSelectedClusters = () => this.setState({selectedClusters: []})
 
+  componentDidMount() {
+    this.props.clearSearch()
+    this.props.clearSearchFilters()
+    this.props.setSearchSort({sortBy: 'default'})
+    this.props.setSearchFilters({location: ChallengeLocation.intersectingMapBounds})
+    this.props.clearMapBounds()
+  }
```
Comment (jschwarz2030) [inline src/components/ChallengePane/ChallengePane.js:115]: how is this related to metrics v2?  if it's not and we're just fixing a defect, we should create a new PR

File: src/components/HOCs/WithSearchResults/WithSearchResults.js [hunk 4]
```diff
@@ -113,4 +132,4 @@ export default (WrappedComponent, searchName, itemsProp, outputProp, searchFunct
     WithSearchResults(WrappedComponent, searchName, itemsProp, outputProp),
     searchName,
     searchFunction
-  )
+  )
```
Comment (jschwarz2030) [inline src/components/HOCs/WithSearchResults/WithSearchResults.js:135]: please revert this file and create a new HOC specifically for metrics.  too many conditions just for admin and too easy to create regressions this way.

File: src/components/SuperAdmin/WithMetricsFilter.js [hunk 5]
```diff
@@ -1,41 +1,117 @@
...
+/**
+ * Deals with superadmin query routes for super admin metrics.
+ * Toggle filter logic here as well.
+ */
 const WithMetricsFilter = function(WrappedComponent) {
   return class extends Component {
     render() {
       const params = queryString.parse(this.props.location.search)
-
+      const tab = params['tab']
```
Comment (jschwarz2030) [inline src/components/SuperAdmin/WithMetricsFilter.js:12]: remove space at end

File: src/components/SuperAdmin/WithSortedUsers.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                               {..._omit(this.props, outputProp)} />
+    }
+  }
+
+  WithSortedUsers.propTypes = {
+    users: PropTypes.array,
+  }
+
+  return WithSortedUsers
+}
```
Comment (jschwarz2030) [inline src/components/SuperAdmin/WithSortedUsers.js:53]: weird spacing.  please use prettier

File: src/services/SuperAdmin/SuperAdminUsers.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export const adminUserEntities = function(state = ADMIN_USERS_INITIAL_STATE, action) {
+  switch (action.type) {
+    case SET_ADMIN_USERS:
+      return { data: action.payload, loadingCompleted: action.loadingCompleted };
+    default:
+      return state;
+  }
+}
```
Comment (jschwarz2030) [inline src/services/SuperAdmin/SuperAdminUsers.js:46]: prettier will add an extra line at the end of files for you.  please use it
